src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+public class Circle extends Shape {
+
+    private StringBuilder message = new StringBuilder();
+    private double roundness;
+
+    public Circle(String name, FigureProducer producer) {
+        super(name);
+        this.roundness = producer.generatorNumber();
+    }
+
+    @Override
+    public double calculateArea() {
+        return (int) (radius() * Math.PI);
```

Comment:
you still cast it to int, so 3.9 will return 3 which is not that good, let's return double, that will be the optimal solution here

Reply:
done!